espnet/asr/pytorch_backend/asr.py

```diff
@@ -986,12 +987,32 @@ def recog(args):
         q_config = {torch.nn.Linear}
 
     if args.quantize_asr_model:
-        assert (
-            "transducer" not in train_args.model_module
-        ), "Quantization of transducer model is not supported yet."
-        logging.info("Use quantized asr model for decoding")
+        logging.info("Use a quantized ASR model for decoding.")
+
+        # It seems quantized LSTM only supports non-packed sequence before torch 1.4.0.
+        # Reference issue: https://github.com/pytorch/pytorch/issues/27963
+        if (
+            torch.__version__ < LooseVersion("1.4.0")
+            and "lstm" in train_args.etype
+            and torch.nn.LSTM in q_config
+        ):
+            raise ValueError(
+                "Quantized LSTM in ESPnet is only supported with torch 1.4+."
+            )
+
+        # Dunno why but weight_observer from dynamic quantized module must have
+        # dtype=torch.qint8 with torch < 1.5 although dtype=torch.float16 is supported.
+        if args.quantize_dtype == "float16" and LooseVersion(
```

Comment:
Can you test this new option in https://github.com/espnet/espnet/blob/master/ci/test_integration_espnet1.sh?

Reply:
Sorry, I'm a bit confused. If I add an integration test for `quantize-dtype: float16`, CI will fail because `ValueError `will be raised?
Ultimately, I would need to add a unit test but we don't directly test `recog(...)` from `asr.py`...

Reply:
I see. So let's go pytorch 1.5+ for the integration test? @sw005320 

I agree that the unittest is very useful, and be careful with pytest.timeout.

Reply:
> CI will fail because ValueError will be raised?

Shall we omit pytorch 1.5- in the CI? @sw005320 

> unittest

Sounds good, and be careful with pytest.timeout.

Reply:
> Sounds good, and be careful with pytest.timeout

Oh btw, how can I be sure which value I should assign to `pytest.mark.execution_timeout(...)`? I mean, I'm setting it up according to my local settings but CI rely on different hardware.

Reply:
Because it is a kinda integration test, you can set a large value enough to pass CI. For example, 50 sec is safe:
https://github.com/espnet/espnet/blob/08feae5bb93fa8f6dcba66760c8617a4b5e39d70/test/espnet2/tasks/test_abs_task.py#L108-L109